### PR TITLE
Bp tree comments and tests

### PR DESCRIPTION
**Pull request by oliviagonzalez**

### CODE REVIEW

File: BPTree.java [hunk 1]
```diff
@@ -454,6 +480,37 @@ void rebalance(InternalNode prev) {
...
+         */
+        Node split() {
+        	
+        	InternalNode sibling = new InternalNode();
+        	int halfPointKey = (keys.size())/2;
+        	int halfPointChild = (children.size())/2;
+        	
+        	//give everything to left of middle key to sibling
+        	for (int i = 0; i < halfPointKey; i++) {
+        		sibling.keys.add(keys.remove(0));
```
Comment (nisler) [inline BPTree.java]: are you only removing the first key or is this supposed to be .remove(i)   ?

File: BPTree.java [hunk 2]
```diff
@@ -454,6 +480,37 @@ void rebalance(InternalNode prev) {
...
+        	}
+        	
+        	//now distribute children
+        	//sibling gets everything up to middle
+        	for (int k = 0; k < halfPointChild; k++) {
+        		sibling.children.add(children.remove(0));
```
Comment (nisler) [inline BPTree.java]: same as above

File: BPTree.java [hunk 3]
```diff
@@ -454,6 +480,37 @@ void rebalance(InternalNode prev) {
...
+        	}
+        	
+        	if (branchingFactor % 2 == 0) {
+        		sibling.children.add(children.remove(0));
```
Comment (nisler) [inline BPTree.java]: same as above

File: BPTree.java (hunk 1 above)
Comment (gcottiero) [reply BPTree.java]: This is intentional, I'm removing the first key

File: BPTree.java (hunk 2 above)
Comment (gcottiero) [reply BPTree.java]: intentional

File: BPTree.java (hunk 3 above)
Comment (gcottiero) [reply BPTree.java]: also intentional

**Pull request merged**